src/daemon/ca/certauth.rs

```diff
@@ -986,7 +986,21 @@ impl CertAuth {
         if !self.parent_known(&parent) {
             Err(Error::CaParentUnknown(self.handle.clone(), parent))
         } else {
-            Ok(vec![CaEvtDet::parent_removed(&self.handle, self.version, parent)])
+            let mut event_details = vec![];
+
+            for (rcn, rc) in &self.resources {
+                if rc.parent_handle() == &parent {
+                    event_details.push(CaEvtDet::ResourceClassRemoved {
+                        resource_class_name: rcn.clone(),
+                        parent: parent.clone(),
+                        revoke_requests: vec![], // We will do a best effort revoke request, but not triggered through this event
```

Comment:
Does this comment agree with the that in `06_ca_manager.md`?
> Note that if the CA actively removes a parent, it will pro-actively send revocation requests for all its keys first, and we do not trigger this task.

Reply:
ah good catch, will update doc. We do, now, remove the resource classes first because this will trigger withdrawing the objects.